### PR TITLE
Add runtime deps tracker interface and related definitions 

### PR DESCRIPTION
**Pull request by pmvald**
The types and interfaces related to the runtime deps tracker are added. Also an empty implementation is added to be completed in follow up PRs (after the interfaces are confirmed in this PR).

The added files are not used anywhere, so the change should not affect anything in anyway.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
@pmvald looks like it will require rebase after recent merges.

**Comment by pmvald** [comment]
> @pmvald looks like it will require rebase after recent merges.

Done!

File: packages/core/src/render3/deps_tracker/api.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+/**
+ * Public API for runtime deps tracker (RDT).
+ *
+ * All downstream tools should only use these methods.
+ */
+export interface DepsTrackerApi {
+  /**
+   * Computes the component dependencies, i.e., a set of components/directive/pipes that could be
+   * present in the component's template (This set might contain elements not present in the
```
Comment (devversion) [inline packages/core/src/render3/deps_tracker/api.ts:50]:
```suggestion
   * present in the component's template (This set might contain directives/components not necessarily used in the
```

**Review by devversion — APPROVED** [review]

**Comment by pmvald** [comment]
@devversion If we want to use this api for JIT as well then we need some more stuff. I added a new commit to this PR to address the JIT specific needs. See if it makes sense. The previous commit is as it is.

**Review by devversion — APPROVED** [review]
Seems reasonable to me. LGTM. I think we agreed on using the deps tracker for JIT as well

**Review by dylhunn — APPROVED** [review]
reviewed-for: fw-core

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 7d7fc48e58f51e574f3a5c1174cd7d95b515ddc8.

**Pull request closed without merging**